### PR TITLE
Droth 361 linear assets history number or textual value

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

**Comment by oappicgi** [comment]
Looks ok to me. Thanks for the descriptive comments.

**Comment by marquesfm** [comment]
Ok to me.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala [hunk 1]
```diff
@@ -511,6 +511,44 @@ trait LinearAssetOperations {
...
+    //Create New Asset
+    val newAssetIDcreate = createWithoutTransaction(oldAsset.typeId, oldAsset.linkId, valueToUpdate, oldAsset.sideCode, oldAsset.startMeasure, oldAsset.endMeasure, username, vvhClient.createVVHTimeStamp(5))
+
+    //Update modifiedBy, modifiedAt and update createdBy, createdDateTime to have the same createdBy/createdDate as old asset
+    val assetsUpdated = Queries.updateAssetModifiedByCopyOldOne(newAssetIDcreate, oldAsset.createdBy,
+      oldAsset.createdDateTime, username).first
+
+    //Update modifiedBy from LRM_POSITION table to have the same modifiedBy as old asset
+    val assetsLrmPosUpdated = Queries.updateLrmPositionModifiedByCopyOldOne(newAssetIDcreate, oldAsset.geomModifiedDate).first
+
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: I belive we can change the createWithoutTransaction to get the modifiedby and modifiedDate. So like this we don't need to have 2 extra updates to the database. 
And i belive we don't need to change the modified date in the old lrm_position.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/masstransitstop/oracle/queries.scala [hunk 2]
```diff
@@ -73,6 +73,19 @@ object Queries {
...
 
+  def updateAssetModifiedByCopyOldOne(assetId: Long, oldCreatedBy: Option[String], oldCreatedDateTime: Option[DateTime],
+                                      updater: String) =
+    sqlu"""
+      update asset set modified_by = $updater, modified_date = CURRENT_TIMESTAMP,
+          created_by = $oldCreatedBy, created_date = $oldCreatedDateTime where id = $assetId
+    """
+
+  def updateLrmPositionModifiedByCopyOldOne(assetId: Long, oldModifiedDateLrmPos: Option[DateTime]) =
+    sqlu"""
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/masstransitstop/oracle/queries.scala]: You can delete this after my first comment.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/masstransitstop/oracle/queries.scala [hunk 3]
```diff
@@ -73,6 +73,19 @@ object Queries {
       update asset set modified_by = $updater, modified_date = CURRENT_TIMESTAMP where id = $assetId
     """
 
+  def updateAssetModifiedByCopyOldOne(assetId: Long, oldCreatedBy: Option[String], oldCreatedDateTime: Option[DateTime],
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/masstransitstop/oracle/queries.scala]: You can delete this after my first comment.

**Comment by joanacoelho** [comment]
Looks ok

**Pull request merged**